Add g:help_example_languages to support language annotation highlighting (#16238)
vim/vim: runtime/doc/helphelp.txt

at the end of the line before the block and a less than (<) character as the
first non-blank on a line following the block.  Any line starting in column 1
also implicitly stops the block of ex-commands before it.  E.g. >
	function Example_Func()
	  echo "Example"

> 	endfunction

Reviewer: why the indentation change?

Author: I have read the other example indentation.

It seems indented by tab.

<

To add annotation in the block, place the annotation (ex: "lua") after a
greater than (>) character.  E.g: >lua
	print("hello")

> <

Reviewer: ```suggestion
<
Note: uses lua syntax highlighting, if "lua" is in |g:help_example_languages|.
```

> In default, it's possible to add Vim syntax highlighting support to code

Reviewer: ```suggestion
It's possible to add Vim syntax highlighting support to code
```

examples.  E.g: >vim
	function Example_Func()
	  echo "Example"
	endfunction
<
						 *g:help_example_languages*
If you want to change the syntax highlighting in the block, you can
change it like this: >

> 	:let g:help_example_languages = #{ vim: 'vim', sh: 'bash' }

Reviewer: Perhaps it should be named: `g:help_import_languages` ?

Author: Hm...  It seems from implementation name.
`help_example_languages` means it describes the example languages.

> The key is language annotation and the value is syntax name.

Reviewer: ```suggestion
The key is the annotation marker and the value is the 'syntax' name.
```

> 

Reviewer: When not configured, help files support Vim script highlighting only.

Author: OK.

Reviewer 2: - The flow from the previous line seems strange.
- Exceeds 80 columns.

at the end of the line before the block and a less than (<) character as the
first non-blank on a line following the block.  Any line starting in column 1
also implicitly stops the block of ex-commands before it.  E.g. >
	function Example_Func()
	  echo "Example"
	endfunction
<

To add annotation in the block, place the annotation (ex: "lua") after a
greater than (>) character.  E.g: >lua
	print("hello")
<
Note: uses lua syntax highlighting, if "lua" is in |g:help_example_languages|.
It's possible to add Vim syntax highlighting support to code
examples.  E.g: >vim
	function Example_Func()
	  echo "Example"
	endfunction
<
						 *g:help_example_languages*
If you want to change the syntax highlighting in the block, you can
change it like this: >
	:let g:help_example_languages = #{ vim: 'vim', sh: 'bash' }
The key is the annotation marker name and the value is the 'syntax' name.
When not configured, help files support Vim script highlighting only.
Note: When setting "g:help_example_languages", if you do not add "vim", Vim

> script syntax highlighting will not be available.

Reviewer: ```suggestion
When setting "g:help_example_languages" to an empty value, syntax highlighting for embedded languages will be disabled.
```

> When setting "g:help_example_languages" to an empty value, syntax highlighting for embedded languages will be disabled.

Reviewer 2: This was posted to vim_dev, but it doesn't seem to be reflected on GitHub, so I'm rewriting it.

- The flow from the previous line seems strange.
- Exceeds 80 columns.

Author: > Exceeds 80 columns.

Thanks.

Author: > The flow from the previous line seems strange.

I have updated the documentation.
